client/signup/steps/plans/index.jsx

```diff
@@ -246,9 +152,9 @@ export class PlansStep extends Component {
 					hideEcommercePlan={ this.shouldHideEcommercePlan() }
 					isInSignup={ true }
 					isLaunchPage={ isLaunchPage }
-					intervalType={ this.getIntervalType() }
-					onUpgradeClick={ this.onSelectPlan }
-					domainName={ this.getDomainName() }
+					intervalType={ getIntervalType() }
+					onUpgradeClick={ this.onSelectPlan() }
```

Comment:
I think a lot of the callbacks that are returned can be removed if we switch to the approach below

```suggestion
					onUpgradeClick={ (cartItem)=> this.onSelectPlan(planProperties, cartItem) }
```

Reply:
That's a nice solution, I'll give it a shot